src/mds/MDSRank.cc

```diff
@@ -818,14 +818,15 @@ void MDSRank::_advance_queues()
 {
   assert(mds_lock.is_locked_by_me());
 
-  while (!finished_queue.empty()) {
+  if (!finished_queue.empty()) {
     dout(7) << "mds has " << finished_queue.size() << " queued contexts" << dendl;
-    dout(10) << finished_queue << dendl;
-    decltype(finished_queue) ls;
-    ls.swap(finished_queue);
```

Comment:
I think it would be preferable to keep `ls.swap` here to so that any buggy context which adds to the finished_queue doesn't cause an infinite loop.

Reply:
'auto fin = finished_queue.front()' is required by queue_waiters_front()


if buggy context keeps adding another  buggy context to finished_queue,  it's infinite loop no matter keep "ls.swap" or not